msal/src/main/java/com/microsoft/identity/client/NativeAuthPublicClientApplicationConfiguration.kt

```diff
@@ -0,0 +1,204 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client
+
+import android.text.TextUtils
+import com.google.gson.annotations.SerializedName
+import com.microsoft.identity.client.configuration.AccountMode
+import com.microsoft.identity.client.exception.MsalClientException
+import com.microsoft.identity.common.java.authorities.CIAMAuthority
+import com.microsoft.identity.common.java.authorities.NativeAuthCIAMAuthority
+import com.microsoft.identity.common.java.logging.LogSession
+import com.microsoft.identity.common.java.logging.Logger
+import lombok.Getter
+import lombok.experimental.Accessors
+import java.io.Serializable
+
+/**
+ * Extends [PublicClientApplicationConfiguration] to add a few extra fields specific to Native Auth
+ * app implementation.
+ */
+@Accessors(prefix = ["m"])
+@Getter
+class NativeAuthPublicClientApplicationConfiguration :
+    PublicClientApplicationConfiguration(),
+    Serializable {
+    companion object {
+        private val TAG = NativeAuthPublicClientApplicationConfiguration::class.java.simpleName
+        private val VALID_CHALLENGE_TYPES = listOf("password", "oob", "redirect")
+    }
+
+    private object NativeAuthSerializedNames {
+        const val CHALLENGE_TYPES = "challenge_types"
+        const val USE_REAL_AUTHORITY = "useRealAuthority"
```

Comment:
Let's follow one naming pattern for the values of Serialized Names (underscores vs camel case). I think underscore is used in most places so let's stick to that.